Gravity/Gravity/Base/BaseDto.cs

```diff
@@ -66,18 +65,15 @@ public object GetPropertyValue(string propertyName)
 			return this.GetType().GetProperty(propertyName).GetValue(this, null);
 		}
 
-		// TODO: Re-work this one to accept selector for the property, not ugly string propertyName
-		public static Guid GetRelativityFieldGuidOfProperty<T>(string propertyName)
-		{
-			var fieldAttribute = typeof(T).GetPublicProperties()
-				.FirstOrDefault(property => property.Name.Equals(propertyName, StringComparison.OrdinalIgnoreCase))?
-				.GetCustomAttribute<RelativityObjectFieldAttribute>();
-			
-
-			return fieldAttribute?.FieldGuid ?? new Guid();
-		}
+        public static T2 GetCustomAttributeOfProperty<T, T2>(string propertyName, Func<RelativityObjectFieldAttribute, T2> relativityObjectFieldAttributSelector)
```

Comment:
As noted in Slack, this is close. What should remove the selector parameter, and put the attribute in as a generic type, like here: https://gravitydev.slack.com/archives/C7Q698BK5/p1521130912000324

Reply:
Thanks for your guidance, think it is where it needs to be and much better than I had it.